fiftyone/factory/repos/execution_store.py

```diff
@@ -38,15 +33,46 @@ def create_store(self, store_name, permissions=None) -> StoreDocument:
         self._collection.insert_one(store_doc.to_mongo_dict())
         return store_doc
 
+    def has_store(self, store_name):
+        """Checks whether a store with the given name exists in the current
+        context.
+        """
+        result = self._collection.find_one(
+            dict(
+                store_name=store_name,
+                key="__store__",
+                dataset_id=self._dataset_id,
+            ),
+            {},
+        )
+        return bool(result)
+
     def list_stores(self) -> list[str]:
-        """Lists all stores in the execution store."""
-        # ensure that only store_name is returned, and only unique values
-        return self._collection.distinct("store_name")
+        """Lists the stores associated with the current context."""
+        result = self._collection.find(
+            dict(key="__store__", dataset_id=self._dataset_id),
+            {"store_name": 1},
+        )
+        return [d["store_name"] for d in result]
+
+    def count_stores(self) -> int:
+        """Counts the stores associated with the current context."""
+        return self._collection.count_documents(
+            dict(key="__store__", dataset_id=self._dataset_id),
+        )
+
+    def delete_store(self, store_name) -> int:
+        """Deletes the specified store."""
+        result = self._collection.delete_many(
+            dict(store_name=store_name, dataset_id=self._dataset_id)
+        )
+        return result.deleted_count
 
     def set_key(self, store_name, key, value, ttl=None) -> KeyDocument:
-        """Sets or updates a key in the specified store"""
-        now = datetime.datetime.now()
+        """Sets or updates a key in the specified store."""
+        now = datetime.utcnow()
```

Comment:
Good catch. This points at missing true mongo tests which would have caught this when running in CI. Note that I did test these locally, however my mongo had the same timezone.

We should still do the same manual testing now that we've changed this logic. I'll do that before we merge this.